perfkitbenchmarker/linux_benchmarks/aws_dynamodb_ycsb_benchmark.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2016 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Run YCSB benchmark against AWS DynamoDB.
+This benchmark does not provision VMs for the corresponding DynamboDB database.
+The only VM group is client group that sends requests to specifiedDB.
+Before running this benchmark, you have to manually create `usertable` with primaryKey 'user02'.
+TODO: fix _Exists logic.
+TODO: add DAX option.
+TODO: add global table option
+"""
+
+import posixpath
+from perfkitbenchmarker import configs
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import ycsb
+from perfkitbenchmarker.providers.aws import aws_dynamodb
+
+BENCHMARK_NAME = 'aws_dynamodb_ycsb'
+BENCHMARK_CONFIG = """
+aws_dynamodb_ycsb:
+  description: >
+      Run YCSB against AWS DynamoDB.
+      Configure the number of VMs via --ycsb_client_vms.
+  vm_groups:
+    default:
+      vm_spec: *default_single_core
+      vm_count: 1"""
+
+YCSB_BINDING_LIB_DIR = posixpath.join(ycsb.YCSB_DIR, 'dynamodb-binding', 'lib')
+AWS_CREDENTIAL_DIR = '/tmp/AWSCredentials.properties'
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('aws_dynamodb_ycsb_awscredentials_properties',
+                    './AWSCredentials.properties',
```

Comment:
Use variable AWS_CREDENTIAL_DIR at line:43

Reply:
the variable at 43 refers to the location to copy the credentials for the vm's to use, and is used at 131 to direct ycsb to the creds, as well as 161 to with the action to copy them up to the vm.  the flag at 46 is used to tell pkb where to copy from.